pkg/cephmgr/client/filesystem.go

```diff
@@ -0,0 +1,149 @@
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
+type CephFilesystem struct {
+	Name           string   `json:"name"`
+	MetadataPool   string   `json:"metadata_pool"`
+	MetadataPoolID int      `json:"metadata_pool_id"`
+	DataPools      []string `json:"data_pools"`
+	DataPoolIDs    []int    `json:"data_pool_ids"`
+}
+
+type CephFilesystemDetails struct {
+	ID     int    `json:"id"`
+	MDSMap MDSMap `json:"mdsmap"`
+}
+
+type MDSMap struct {
+	FilesystemName string             `json:"fs_name"`
+	Enabled        bool               `json:"enabled"`
+	Root           int                `json:"root"`
+	TableServer    int                `json:"tableserver"`
+	MaxMDS         int                `json:"max_mds"`
+	In             []int              `json:"in"`
+	Up             map[string]int     `json:"up"`
+	Failed         []int              `json:"failed"`
+	Damaged        []int              `json:"damaged"`
+	Stopped        []int              `json:"stopped"`
+	Info           map[string]MDSInfo `json:"info"`
+}
+
+type MDSInfo struct {
+	GID     int    `json:"gid"`
+	Name    string `json:"name"`
+	Rank    int    `json:"rank"`
+	State   string `json:"state"`
+	Address string `json:"addr"`
+}
+
+func ListFilesystems(conn Connection) ([]CephFilesystem, error) {
+	cmd := map[string]interface{}{"prefix": "fs ls"}
+	buf, err := ExecuteMonCommand(conn, cmd, "list filesystems")
+	if err != nil {
+		return nil, fmt.Errorf("failed to list filesystems: %+v", err)
+	}
+
+	var filesystems []CephFilesystem
+	err = json.Unmarshal(buf, &filesystems)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal failed: %+v.  raw buffer response: %s", err, string(buf))
+	}
+
+	return filesystems, nil
+}
+
+func GetFilesystem(conn Connection, fsName string) (*CephFilesystemDetails, error) {
+	cmd := map[string]interface{}{
+		"prefix":  "fs get",
+		"fs_name": fsName,
+	}
+	buf, err := ExecuteMonCommand(conn, cmd, fmt.Sprintf("fs get %s", fsName))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get file system %s: %+v", fsName, err)
+	}
+
+	var fs CephFilesystemDetails
+	err = json.Unmarshal(buf, &fs)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal failed: %+v.  raw buffer response: %s", err, string(buf))
+	}
+
+	return &fs, nil
+}
+
+func CreateFilesystem(conn Connection, fsName, metadataPool, dataPool string) error {
+	cmd := map[string]interface{}{
+		"prefix":   "fs new",
+		"fs_name":  fsName,
+		"metadata": metadataPool,
+		"data":     dataPool,
+	}
+	_, err := ExecuteMonCommand(conn, cmd, fmt.Sprintf("enabling ceph fs %s", fsName))
```

Comment:
I think ExecuteMonCommand on its own still has merit, I don't think it's strictly necessary to enforce strong type wrappers around all mon commands.